Utils/Services/ImageFetcher.swift

```diff
@@ -11,6 +11,12 @@ import ReactiveSwift
 
 public protocol ImageFetcherType {
     
+    /**
+     Fetches an image asynchronously from a URL, returning a value or an error in the SignalProducer when done.
+     - parameter imageURL: the url of the image to fetch.
+     - parameter session: the URLSession to use to make the request. Default: URLSession.shared.
```

Comment:
This default is for the `ImageFetcher`, not any ImageFetcherType. So I would move it to the ImageFetcher class:
```
/**
Class for fetching images through ImageFetcherType protocol.
It uses the shared URLSession by default.
*/
public class ImageFetcher: ImageFetcherType {
```